src/Services/Azure/DataExplorer/DataExplorerService.cs

```diff
@@ -0,0 +1,315 @@
+
+        using var queryProvider = KustoClientFactory.CreateCslAdminProvider(kcsb);
+        var reader = await queryProvider.ExecuteControlCommandAsync(databaseName, ".show tables", null);
+        var result = new List<string>();
+        while (reader.Read())
+        {
+            result.Add(reader["TableName"].ToString()!);
+        }
+        return result;
+    }
+
+    public async Task<List<JsonDocument>> GetTableSchema(
+        string subscriptionId,
+        string clusterName,
+        string databaseName,
+        string tableName,
+        string? tenant = null,
+        AuthMethod? authMethod = AuthMethod.Credential,
+        RetryPolicyArguments? retryPolicy = null)
+    {
+        var clusterUri = await GetClusterUri(subscriptionId, clusterName, tenant, retryPolicy);
+        return await GetTableSchema(clusterUri, databaseName, tableName, tenant, authMethod, retryPolicy);
+    }
+
+    public async Task<List<JsonDocument>> GetTableSchema(
+        string clusterUri,
+        string databaseName,
+        string tableName,
+        string? tenant = null,
+        AuthMethod? authMethod = AuthMethod.Credential,
+        RetryPolicyArguments? retryPolicy = null)
+    {
+        ValidateRequiredParameters(clusterUri, databaseName, tableName);
+        var kcsb = await CreateKustoConnectionStringBuilder(clusterUri, authMethod, null, tenant);
+        using var queryProvider = KustoClientFactory.CreateCslAdminProvider(kcsb);
+        var command = $".show table {tableName} schema as json";
+        var reader = await queryProvider.ExecuteControlCommandAsync(databaseName, command, null);
+        var result = new List<JsonDocument>();
+        while (reader.Read())
+        {
+            var jsonString = reader["Schema"].ToString()!;
+            var json = JsonDocument.Parse(jsonString);
+            result.Add(json);
+            
+        }
+        return result;
+    }
+
+    public async Task<List<JsonDocument>> QueryItems(
+        string subscriptionId, 
+        string clusterName, 
+        string databaseName, 
+        string query, 
+        string? tenant = null, 
+        AuthMethod? authMethod = AuthMethod.Credential, 
+        RetryPolicyArguments? retryPolicy = null)
+    {
+        ValidateRequiredParameters(subscriptionId, clusterName, databaseName, query);
+
+        string clusterUri = await GetClusterUri(subscriptionId, clusterName, tenant, retryPolicy);
+
+        return await QueryItems(clusterUri, databaseName, query, tenant, authMethod, retryPolicy);
+    }
+
+    private async Task<string> GetClusterUri(string subscriptionId, string clusterName, string? tenant, RetryPolicyArguments? retryPolicy)
+    {
+        var cluster = await GetCluster(subscriptionId, clusterName, tenant, retryPolicy) ?? throw new Exception($"Data Explorer cluster '{clusterName}' not found in subscription '{subscriptionId}'.");
+
+        if (!cluster.RootElement.TryGetProperty("ClusterUri", out var clusterUriElement))
```

Comment:
did you test this? IIRC, the property is "uri"

Reply:
I tested it, see [Azure.ResourceManager.Kusto](https://learn.microsoft.com/en-us/dotnet/api/azure.resourcemanager.kusto?view=azure-dotnet)